aptos-move/move-examples/parallel_accounts/sources/parallel_accounts_demo.move

```diff
@@ -0,0 +1,129 @@
+
+    /// Allows owner to do transfer actions
+    struct TransferCapability has drop, store { account: address }
+
+    /// Creates a resource account and stores the capability
+    public entry fun create_resource_account_and_store_cap(sender: &signer, seed: vector<u8>) {
+        let (resource_signer, signer_cap) = create_demo_resource_account(sender, seed);
+
+        // Store
+        move_to(&resource_signer, TransferResource {
+            signer_cap
+        });
+
+        move_to(sender, TransferAdmin {
+            admin_cap: AdminCapability {
+                account: address_of(&resource_signer)
+            },
+        });
+    }
+
+    /// Adds a transfer cap to an account, transaction must be multi-agent
+    public entry fun delegate_transfer_cap(admin: &signer, delegate: &signer) acquires TransferAdmin {
+        let admin_account = address_of(admin);
+        assert!(exists<TransferAdmin>(admin_account), ENO_ADMIN_CAP_FOUND);
+        let admin = borrow_global<TransferAdmin>(admin_account);
+        move_to(delegate, TransferDelegate {
+            transfer_cap: TransferCapability {
+                account: admin.admin_cap.account
+            }
+        });
+    }
+
+    /// Removes transfer cap from an account
+    public entry fun revoke_transfer_cap(admin: &signer, delegate: address) acquires TransferDelegate {
+        let admin_account = address_of(admin);
+        assert!(exists<TransferAdmin>(admin_account), ENO_ADMIN_CAP_FOUND);
+        assert!(exists<TransferDelegate>(delegate), ENO_TRANSFER_CAP_FOUND);
+        move_from<TransferDelegate>(delegate);
+    }
+
+    /// Transfers coins with the TransferCapability
+    public entry fun transfer_coins<CoinType>(
+        payer: &signer,
+        receiver: address,
+        amount: u64
+    ) acquires TransferResource, TransferDelegate {
+        let resource_account = create_signer_for_delegate(payer);
+        aptos_account::transfer_coins<CoinType>(&resource_account, receiver, amount)
+    }
+
+    public entry fun offer_token<CoinType>(
+        payer: &signer,
+        receiver: address,
+        creator: address,
+        collection: String,
+        name: String,
+        property_version: u64,
+        amount: u64
+    ) acquires TransferResource, TransferDelegate {
+        let resource_account = create_signer_for_delegate(payer);
+        offer_script(resource_account, receiver, creator, collection, name, property_version, amount)
+    }
+
+    #[view]
+    /// Retrieves the resource account address used with this contract
+    public fun resource_account_address(payer: address, seed: vector<u8>): address {
+        let seed = create_seed(seed);
+        create_resource_address(&payer, seed)
+    }
+
+    inline fun create_seed(seed: vector<u8>): vector<u8> {
+        vector::append(&mut seed, DEMO_SEED);
+        seed
+    }
+
+    inline fun create_demo_resource_account(payer: &signer, seed: vector<u8>): (signer, SignerCapability) {
```

Comment:
Why does this need to be an inline function?

Reply:
It doesn't have to be, but what's the tradeoff between a private function and an inline right now?